caikit/runtime/servicers/training_management_servicer.py

```diff
@@ -80,6 +82,8 @@ def CancelTraining(self, request, context):  # pylint: disable=unused-argument
                 state=model_future.get_info().status,
                 reasons=reasons,
             ).to_proto()
+        except CaikitCoreException as err:
```

Comment:
I think the one thing missing here is handling the issue where `model_future.get_info()` can return a 404 after `model_future.cancel()`. Would we want to catch that and just return `CANCELLED`?

It does feel a bit wonky to handle here since that's a quirk of some remote trainers, and we would have to write a mock or something to behave this way since the local trainer won't. Did you wanna tackle now or in a future PR?